src/commands/sourcemaps/resolve.rs

```diff
@@ -0,0 +1,107 @@
+use std::fs;
+use std::path::PathBuf;
+
+use anyhow::{format_err, Result};
+use clap::{Arg, ArgMatches, Command};
+use sourcemap::{DecodedMap, Token};
+
+use crate::utils::args::validate_int;
+
+pub fn make_command(command: Command) -> Command {
+    command
+        .about("Resolve sourcemap for a given line/column position.")
+        .arg(
+            Arg::new("path")
+                .value_name("PATH")
+                .help("The sourcemap to resolve."),
+        )
+        .arg(
+            Arg::new("line")
+                .long("line")
+                .short('l')
+                .value_name("LINE")
+                .validator(validate_int)
+                .help("Line number for minified source."),
+        )
+        .arg(
+            Arg::new("column")
+                .long("column")
+                .short('c')
+                .value_name("COLUMN")
+                .validator(validate_int)
+                .help("Column number for minified source."),
+        )
+}
+
+/// Retrurns the zero indexed position from matches
+fn lookup_pos(matches: &ArgMatches) -> Option<(u32, u32)> {
+    Some((
+        matches
+            .value_of("line")
+            .map_or(0, |x| x.parse::<u32>().unwrap() - 1),
+        matches
+            .value_of("column")
+            .map_or(0, |x| x.parse::<u32>().unwrap() - 1),
+    ))
+}
+
+fn print_token(token: &Token<'_>) {
+    if let Some(name) = token.get_name() {
+        println!("  name: {:?}", name);
+    } else {
+        println!("  name: not found");
+    }
+    if let Some(source) = token.get_source() {
+        println!("  source file: {:?}", source);
+    } else {
+        println!("  source file: not found");
+    }
+    println!("  source line: {}", token.get_src_line());
+    println!("  source column: {}", token.get_src_col());
+    println!("  minified line: {}", token.get_dst_line());
+    println!("  minified column: {}", token.get_dst_col());
+    if let Some(view) = token.get_source_view() {
+        println!("  source code:");
+        for offset in &[-2, -1, 0, 1, 2] {
+            let line = token.get_src_line() as isize + offset;
+            if line < 0 {
+                continue;
+            }
+            if let Some(line) = view.get_line(line as u32) {
+                println!("    {}", line.trim());
+            }
+        }
+    } else if token.get_source_view().is_none() {
+        println!("  cannot find source");
+    } else {
+        println!("  cannot find source line");
+    }
+}
+
+pub fn execute(matches: &ArgMatches) -> Result<()> {
+    let sourcemap_path = matches
+        .value_of("path")
+        .ok_or_else(|| format_err!("Sourcemap not provided"))?;
+
+    let sm = sourcemap::decode_slice(&fs::read(PathBuf::from(sourcemap_path))?)?;
+
+    let ty = match sm {
+        DecodedMap::Regular(..) => "regular",
+        DecodedMap::Index(..) => "indexed",
+        DecodedMap::Hermes(..) => "hermes",
```

Comment:
do we just make returning the type a method on `DecodedMap` so we don't have to come back and update this match statement when we add new one's in the future?

Reply:
We'd have to change that directly in `rust-sourcemap`, but it changes so rarely that I'd not bother tbh.